ignite/base/events_driven.py

```diff
@@ -0,0 +1,319 @@
+import functools
+import logging
+import weakref
+from collections import defaultdict
+from typing import Any, Callable, Dict, List, Mapping, Optional, Union
+
+from ignite.engine.events import CallableEventWithFilter, EventEnum, Events, EventsList, RemovableEventHandle
+from ignite.engine.utils import _check_signature
+
+
+class EventsDriven:
+    """Base class for events-driven engines without state. This class is mainly
+    respoinsible for registring events and fire events, also keep track of
+    the allowed events, and how many time they have been fired.
+
+    Attributes:
+        last_event_name: last event name triggered by the engine.
+
+    Examples:
```

Comment:
This does not look good:
- https://deploy-preview-2107--pytorch-ignite-preview.netlify.app/generated/ignite.base.events_driven.eventsdriven#ignite.base.events_driven.EventsDriven

Reply:
I agree that doesn't look very neat, but that's the same way we render examples in the entire codebase, should I find a way to fix it here? or leave it?

Reply:
We may do that later everywhere. Could you please open an issue about that. Thanks !